### PR TITLE
Got started on input lane and code to store input lane options, etc.

### PR DESCRIPTION
**Pull request by jscarlsson**

### CODE REVIEW

File: App/components/InputLane.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Text, View } from 'react-native';
+import PropTypes from 'prop-types';
+import { ScrollView } from 'react-native-gesture-handler';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+
+// this is maybe temporary, I feel like this should be in a JSON file somewhere, which maybe also interacts with TrackingSettings
```
Comment (sanlil) [inline App/components/InputLane.jsx:7]: What do you think about something like this format?

```
trackingLanes = [
  {
    title: 'tracking.cervical_mucus.title'
    type: MULTI_CHOICE,
    fields: [
      {
        name: 'tracking.cervical_mucus.sticky',
        icon: 'assets/img/cm_sticky.png'
      },
      {
        name: 'tracking.cervical_mucus.creamy',
        icon: 'assets/img/cm_creamy.png'
      },
      {
        name: 'tracking.cervical_mucus.eggwhite',
        icon: 'assets/img/cm_eggwhite.png'
      }
    ]
    color: 'theme.colors.teal.light',
  },
  {
    title: 'tracking.bleeding.title'
    type: LEVEL,
    levels: ['tracking.bleeding.light', 'tracking.bleeding.mid', 'tracking.bleeding.heavy']
    fields: [
      {
        name: 'tracking.bleeding.menstruation',
        icon: 'assets/img/period.png'
      },
      {
        name: 'tracking.bleeding.spoting',
        icon: 'assets/img/spotting.png'
      }
    ],
    color: 'theme.colors.burgundy.light'
  }
]
```

Comment (sanlil) [reply App/components/InputLane.jsx:7]: But the component shouldn't have to interact directly with the JSON, but get the data for one of the input lanes through the props

Comment (sanlil) [reply App/components/InputLane.jsx:7]: But this component would of course not interact with the full JSON, but get the relevant parts as props. So `cmStates` and `bleedingStates` would come from `trackingLanes.fields`

File: App/screens/TrackingSettings.jsx [hunk 2]
```diff
@@ -4,48 +4,23 @@ import { IconButton, Switch } from 'react-native-paper';
...
-    'Alcohol',
-    'Coffee',
-    'Sugar',
-  ];
-
-  // Create arrays holding the different option views
-
   const observationSettings = generateSettingsRows(
-    observationNames,
+    Object.keys(trackingSettings.observationNames),
```
Comment (david8611) [inline App/screens/TrackingSettings.jsx:13]: Using each setting key value from the json-file.

**Comment by david8611** [comment]
App/appData/tracking-lanes.json

File: App/appData/tracking-lanes.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  },
+  "activityNames": {
+    "Sex": {},
+    "Sleep": {},
+    "Exercise": {},
+    "Alcohol": {},
+    "Coffee": {},
+    "Sugar": {}
+  }
+}
```
Comment (david8611) [inline App/appData/tracking-lanes.json:63]: Added a json-file to structure and store each key-value for each input-lane.

File: App/components/InputLane.jsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+function generateInputLane(stateList, laneKey, currentDate) {
+  const [laneState, setLaneState] = useState('');
+  const InputOption = ({ option, color }) => {
+    const [isSelected, changeSelected] = useState(false);
+    // Here we need to be able to only save the current state from the current date and not for all dates.
+    useEffect(() => {
+      storeInput(laneKey, currentDate, laneState);
+      checkIsSelected(laneKey, currentDate, option, changeSelected);
+    }, [laneState]);
```
Comment (david8611) [inline App/components/InputLane.jsx:82]: I need some help here to only store the inputs for each date. Currently there is a bug, storing the same value across all dates for some reason, I do have different keys for each stored value.

**Comment by david8611** [comment]
Right now I believe we have enough for some usability testing with input tracking lanes based on the trackingSettings with each input stored locally. I currently have limited time improving this version with some better design of the input lanes this week. It would be great if someone else could take a look at the design implementation (style, etc) of the input lanes. I also would like someone to help me solve the issue storing only each input for the specific date. The documentation is also lacking due to limited time atm.

**Comment by david8611** [comment]
Added support for selecting several input lane options by storing an array of the different input lane options.

File: App/components/InputLane.jsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+/**
+ * Generate each input lane-row.
+ * @param {*} stateList
+ * @param {*} laneKey
+ * @param {*} currentDate
+ * @returns
+ */
+function generateInputLane(stateList, laneKey, currentDate) {
+  const [laneState, setLaneState] = useState([]);
```
Comment (david8611) [inline App/components/InputLane.jsx:119]: Added support for selecting several input lane options by storing an array of the different input lane options.

**Comment by epulls** [comment]
bild
This is a known error and should need its own pull request.

**Review by epulls — APPROVED** [review]

**Review by david8611 — APPROVED** [review]

**Pull request merged**